cmd/otelcol/main.go

```diff
@@ -169,44 +155,79 @@ func checkRuntimeParams() {
-	if config == "" {
-		_, err := os.Stat(defaultDockerSAPMConfig)
-		if err == nil {
-			config = defaultDockerSAPMConfig
+// Set config file path environment variable to specified config file path flag.
+// Set the config file path flag to the specified config file path environment variable if not specified.
+// Set the config file path flag, config file path environment variable to default config file path if the config
+// file path flag, config file path environment variable and config YAML environment variable are not specified.
+func setConfigPath() {
+	configPathFlag := getKeyValue(os.Args[1:], "--config")
+	configPathVar := os.Getenv(configEnvVarName)
+	configYaml := os.Getenv(configYamlEnvVarName)
+
+	if configPathFlag == "" && configPathVar == "" {
+		if configYaml == "" {
+			setConfigPathToDefault()
+		} else {
+			log.Printf("Using environment variable %s for configuration", configYamlEnvVarName)
 		}
-		_, err = os.Stat(defaultLocalSAPMConfig)
-		if err == nil {
-			config = defaultLocalSAPMConfig
+		return
+	}
+
+	if configPathFlag != "" {
+		if configPathVar != "" && configPathVar != configPathFlag {
+			log.Printf("Both %v and '--config' were specified. Overriding %q environment variable value with %q for this session", configEnvVarName, configPathVar, configPathFlag)
 		}
-		if config == "" {
-			log.Fatalf("Unable to find the default configuration file, ensure %s environment variable is set properly", configEnvVarName)
+		os.Setenv(configEnvVarName, configPathFlag)
+
+		if configYaml != "" {
+			log.Printf("Both %s and '--config' were specified. Using '--config' value %s for this session", configYamlEnvVarName, configPathFlag)
 		}
-	} else {
-		// Check if file exists.
-		_, err := os.Stat(config)
-		if err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", config, configEnvVarName)
+
+		log.Printf("Set config to %v", configPathFlag)
+		return
+	}
+
+	if configPathVar != "" {
+		if !contains(os.Args[1:], "--config") {
+			// Inject the command line flag that controls the configuration.
+			os.Args = append(os.Args, "--config="+configPathVar)
+		}
+
+		if configYaml != "" {
+			log.Printf("Both %s and %s were specified. Using %s environment variable value %s for this session", configEnvVarName, configYamlEnvVarName, configEnvVarName, configPathVar)
 		}
+
+		log.Printf("Set config to %v", configPathVar)
+	}
+}
+
+func setConfigPathToDefault() {
+	var path string
+	// Use a default config if no config given; supports Docker and local
+	_, err := os.Stat(defaultDockerSAPMConfig)
+	if err == nil {
+		path = defaultDockerSAPMConfig
+	}
+	_, err = os.Stat(defaultLocalSAPMConfig)
+	if err == nil {
+		path = defaultLocalSAPMConfig
+	}
+	if path == "" {
+		log.Fatalf("Unable to find the default configuration file, ensure %s environment variable is set properly", configEnvVarName)
 	}
 
-	switch config {
+	switch path {
```

Comment:
By pushing this to the
```
	if configPathFlag == "" && configPathVar == "" && configYaml == ""
```
case only env var validation won't occur for default installation paths that use the config env var (all of them iirc).  This should occur separately.